src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h

```diff
@@ -219,12 +219,15 @@ JSON_ENUM_MAPPER(::winrt::Microsoft::Terminal::Settings::Model::FirstWindowPrefe
 
 JSON_ENUM_MAPPER(::winrt::Microsoft::Terminal::Settings::Model::LaunchMode)
 {
-    JSON_MAPPINGS(5) = {
+    JSON_MAPPINGS(8) = {

```

Comment:
I'm a bit conflicted about this. At this point, why not use `JSON_FLAG_MAPPER`? I guess we already introduced `maximizedFocus` as a valid global setting option, so this would be a breaking change.

To fix this, we would have to manually handle `maximizedFocus`, and that just seems like a pain to do right now (with very little benefit tbh). Could you file this as a follow-up please?

- - - - - 

Another thing, what happens if we use one of these in the `"launchMode"` global setting? Does it work as expected? We'll have to update the schema and docs for this too, right?

Reply:
I didn't know about `JSON_FLAG_MAPPER`, I can look into that. 

I haven't tested, but hypothetically if you change the global setting to one of the new options it should launch correctly (since it uses the same code path as restoring), although I don't think it would display in the settings UI properly.

Reply:
Bah, the settings editor does not like the change to `Flags`, I'll fix that.

Reply:
option one, give the user all options: 
![image](https://user-images.githubusercontent.com/6185249/148448804-05abec8f-6e63-4ed0-9c60-c906dbd23fc8.png)

option two, let the user set any of the options in settings.json, but only show the previous options in the ui:
![image](https://user-images.githubusercontent.com/6185249/148449409-b13d0db7-cc85-46ba-9b66-33cae5bbc826.png)

Reply:
bah, I'm ok with option 2. IIRC that means that if you hit save in the ui, the setting won't be overwritten unless you actually change/interact with the setting via the ui? (mind confirming that)

If that's the case, we should be good to go. We'll just treat the extra `launchMode` global setting values as a "secret feature" because the alternative is an unnecessary pain to face haha.

Reply:
Yes, if you hit save without interacting with the launch mode radio buttons it will preserve whatever value you put in the json.